### DISCUSSION TITLE
Godot Breakthrough ideas

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#10314] [General Discussions]
What would be some HUGE Godot additions that would put Godot on the map:

1. Built in full SQL database functionality - Imagine all the power of SQL to handle your game data. Actual relational Database tables instead of importing janky csv files with janky code.

2. Fully working real-time multiplayer with a check box for roll back. Multiplayer is considered an ultra Advanced topic on all 3 major engines (Uni, Unr, God). Imagine if it just worked.

Any more BIG functionalities Godot could add or thoughts on these?

**Comment by AThousandShips** [comment]
There already proposals tracking these two ideas, please look at them directly instead, discussing ideas in multiple places just makes it harder to keep track of ideas

**Reply by shak2** [reply]
I do not see them discussed. Please link to their discussions or tell where to find it please.

Example: [https://github.com/godotengine/godot-proposals/discussions?discussions_q=sql](https://github.com/godotengine/godot-proposals/discussions?discussions_q=sql) returns nothing of value.

**Reply by AThousandShips** [reply]
Not specifically SQL but other database support, like:
* https://github.com/godotengine/godot-proposals/discussions/4609
* https://github.com/godotengine/godot-proposals/issues/2691

For example 

For multiplayer:
* https://github.com/godotengine/godot-proposals/issues/9663
* https://github.com/godotengine/godot-proposals/issues/2821

And exactly:
* https://github.com/godotengine/godot-proposals/issues/8956

But regardless specific ideas should have their own dedicated discussion or proposal

**Reply by shak2** [reply]
I believe there is value in having a place where people can brainstorm bigger ideas without the pressure to create a complete proposal (For example, they might want the feature but lack the technical ability to even begin to comprehend it's implementation). For them, there is currently no way to easily get an idea out there, except now, this one thread, which might catch a few fish who help the Godot project more as their skills develop over time.

We are also missing a way to contribute tiny actionable quick changes that users might feel too lazy to push themselves because to do so is long and convoluted. Like correcting a small spelling or grammatical error in the docs. A regular forum would have a pinned topic for such suggestions which mods can then quickly implement in aggregate.

**Reply by AThousandShips** [reply]
Except having one discussion with multiple things means you lose discussion progress when you discuss it in multiple places, and if someone does feel interested enough in a feature to actually make a proposal the progress will be lost

> Like correcting a small spelling or grammatical error in the docs.

That is different from this which discusses large scale ideas and suggestions, for tracking minor details *on the same subject* a tracker is correct, and we have plenty of those, but they do *not* replace individual issue reports or proposal issues

Just starting a basic discussion with one topic isn't any more work than making one with multiple topics

But in any case this is not really relevant as the second idea discussed has a proposal already, and the first has related ones

Also just see the guide for proposing features:
> Please open one proposal per feature requested. Do not cram multiple feature requests in a single proposal, as this makes it harder to discuss features individually.

**Reply by theraot** [reply]
> I believe there is value in having a place where people can brainstorm bigger ideas without the pressure to create a complete proposal

These discussions are good for this.

---

> Like correcting a small spelling or grammatical error in the docs.

For the documentation, you can open an issue here: https://github.com/godotengine/godot-docs/issues

---

> A regular forum would have a pinned topic for such suggestions which mods can then quickly implement in aggregate.

If what you want is to make a list of features you want (not discussing the implementation), then I think it is useful to link existing proposals or pull requests. That way we can see which proposals get implemented and merged, which get closed or rejected. If you want to discuss the specific implementation, then comment on the proposal or pull request.

With that said, I should point out a couple things:

- A full relational database and a one-click multiplayer solution are not "quick changes" or something to "quickly implement".
- And "quickly implement in aggregate" is a bad idea. 1. A lot of changes in a single pull request makes it harder to review and test, and thus might take longer to merge and might make it easier for bugs to sneak in. 2. If there are multiple such pull requests, merge conflicts can get harder to solve. 3. It makes it harder to revert changes when a feature has bugs but not the others from the same pull request.

---

By the way, you might also be interested in the tags of the issues.

- In the godot repository people can easily jump in to issues with "good first issue" which are usually easy to fix, and "needs testing" which virtually anybody can test (although might need a particular hardware or platform depending on the issue). You might also be interested in "tracker" which are issues that track other issues.
- In the godot-docs repository you also find "good first issue" and "tracker".
- And in the godot-proposals repository you find "implementer wanted" but be aware that these might need specific skill sets to tackle.

**Reply by AThousandShips** [reply]
> A full relational database and a one-click multiplayer solution are not "quick changes" or something to "quickly implement".

Further *most* projects do not need a database system, so this is best handled as a plugin, only really certain types of projects need databases especially these kind of cases, mostly this belongs in servers (which aren't always implemented in Godot in the first place) or other software, most game applications I'd say don't use or need databases

**Reply by theraot** [reply]
@shak2 Something else, we have this: https://godot-proposals-viewer.github.io/ you might find that a easier way to get an idea of what proposal people are most interested in (with the caveat that proposals shared by the core team on social media often get more early attention).
